Source/Mosa.DeviceSystem/Bitmap.cs

```diff
@@ -20,24 +20,21 @@ public static Image CreateImage(byte[] data)
 			var height = stream.Read32(0x16);
 			var dataSectionOffset = stream.Read32(0xA);
 
-			var image = new Image(width, height);
+			var image = new Image(width, height, bpp);
 
-			int x = 0;
-			int y = height - 1;
-
-			for (var i = dataSectionOffset; i < dataSectionOffset + (width * height * bpp); i += bpp)
+			// TODO: Support for other color depths
+			for (uint y = 0; y < height - 1; y++)
```

Comment:
Since the data is stored from bottom row of pixels to top row of pixels, the starting value of y needs to be (height - 1) and keep going until you get to 0 inclusive.

It might be better to change this first loop into a do-while loop, e.g.:
```
uint y = height - 1;
do {
....
--y;
} while (y > 0)
```

Reply:
I prefer the aesthetics of a for loop though. I'll just modify the loop.